client/integration-tests/path/src/test/java/io/quarkiverse/openapi/generator/it/PathTest.java

```diff
@@ -0,0 +1,106 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import jakarta.ws.rs.Path;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class PathTest {
+
+    @Test
+    void testPathGenerated() throws ClassNotFoundException, NoSuchMethodException {
+        String apiClassName = "org.openapi.quarkus.openapi_path_yaml.api.UserResourceApi";
+        String modelClassName = "org.openapi.quarkus.openapi_path_yaml.model.User";
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/users");
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceFind", Integer.class)
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/{id}");
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceFindAll")
+                        .getAnnotation(Path.class))
+                .isNull();
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceAdd", Class.forName(modelClassName))
+                        .getAnnotation(Path.class))
+                .isNull();
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceUpdate", Integer.class, Class.forName(modelClassName))
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/{id}");
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceDelete", Integer.class)
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/{id}");
+    }
+
+    @Test
+    void testPathWithForwardSlashGenerated() throws ClassNotFoundException, NoSuchMethodException {
+        String apiClassName = "org.openapi.quarkus.openapi_path_with_forward_slash_yaml.api.UserResourceApi";
+        String modelClassName = "org.openapi.quarkus.openapi_path_with_forward_slash_yaml.model.User";
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/users");
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceFind", Integer.class)
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/{id}");
+
+        assertThat(
+                Class.forName(apiClassName)
+                        .getMethod("userResourceFindAll")
+                        .getAnnotation(Path.class)
+                        .value())
+                .isEqualTo("/");
```

Comment:
So now, instead of `locahost:8080`, I will have `localhost:8080/`? And will it work both ways? If not, we should not force the slash in the end.

Reply:
The slash in the end is not force. instead of locahost:8080, you will have localhost:8080

It depend on openapi documents

If the document has slashes

https://github.com/quarkiverse/quarkus-openapi-generator/blob/ab24f97a192005132e1d3ee81f41fddc117ebf4f/client/integration-tests/path/src/main/openapi/openapi-path-with-forward-slash.yaml#L12

we will have slashes

https://github.com/quarkiverse/quarkus-openapi-generator/blob/ab24f97a192005132e1d3ee81f41fddc117ebf4f/client/integration-tests/path/src/test/java/io/quarkiverse/openapi/generator/it/PathTest.java#L82

If the document has no slash

https://github.com/quarkiverse/quarkus-openapi-generator/blob/ab24f97a192005132e1d3ee81f41fddc117ebf4f/client/integration-tests/path/src/main/openapi/openapi-path.yaml#L12

No path annotation generated

https://github.com/quarkiverse/quarkus-openapi-generator/blob/ab24f97a192005132e1d3ee81f41fddc117ebf4f/client/integration-tests/path/src/test/java/io/quarkiverse/openapi/generator/it/PathTest.java#L42